src/ts_catalog/continuous_agg.c

```diff
@@ -1486,35 +1498,114 @@ ts_continuous_agg_bucket_width(const ContinuousAgg *agg)
 	return agg->data.bucket_width;
 }
 
+/* Calls one of time_bukcket_ng() versions depending on the arguments. */
+static Datum
+generic_time_bucket_ng(const ContinuousAggsBucketFunction *bf, Datum timestamp)
+{
+	if (strlen(bf->timezone) > 0)
+	{
+		return DirectFunctionCall3(ts_time_bucket_ng_timezone,
+								   IntervalPGetDatum(bf->bucket_width),
+								   timestamp,
+								   CStringGetTextDatum(bf->timezone));
+	}
+	else
+	{
+		return DirectFunctionCall2(ts_time_bucket_ng_timestamp,
+								   IntervalPGetDatum(bf->bucket_width),
+								   timestamp);
+	}
```

Comment:
```suggestion

	return DirectFunctionCall2(ts_time_bucket_ng_timestamp,
								   IntervalPGetDatum(bf->bucket_width),
								   timestamp);
```

Minor simplification!